src/main/java/shef/servlets/UserServlet.java

```diff
@@ -70,19 +75,21 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       boolean isCurrentUser;
 
       if (userService.isUserLoggedIn()) {
+        // ID is referred to as "Name" in Datastore.
+        String id = (String) userKey.getName();
         isCurrentUser = id.equals(userService.getCurrentUser().getUserId());
       } else {
```

Comment:
I think this else if statement is unnecessary. `id.equals(userService.getCurrentUser().getUserId())` will return true if the current user is the same user logged in and false if not, so you don't need an if-else statement to manually set isCurrentUser to false.